src/field_theory/finite.lean

```diff
@@ -9,40 +9,53 @@ import data.zmod.basic
 import linear_algebra.basis
 import ring_theory.integral_domain
 
-universes u v
-variables {K : Type u} {R : Type v}
+/-!
+# Finite fields
 
-open function finset polynomial nat
+This file contains basic results about finite fields.
+Throughout most of this file, `K` denotes a finite field with `q` elements.
 
-namespace finite_field
+## Main results
 
-def field_of_integral_domain [fintype R] [decidable_eq R] [integral_domain R] :
-  field R :=
-{ inv := λ a, if h : a = 0 then 0
-    else fintype.bij_inv (show function.bijective (* a),
-      from fintype.injective_iff_bijective.1 $ λ _ _, (domain.mul_left_inj h).1) 1,
-  mul_inv_cancel := λ a ha, show a * dite _ _ _ = _, by rw [dif_neg ha, mul_comm];
-    exact fintype.right_inverse_bij_inv (show function.bijective (* a), from _) 1,
-  inv_zero := dif_pos rfl,
-  ..show integral_domain R, by apply_instance }
+1. Every finite integral domain is a field (`field_of_integral_domain`).
+2. The unit group of a finite field is a cyclic group of order `q - 1`.
+   (`finite_field.is_cyclic` and `card_units`)
+3. `sum_pow_units`: The sum of `x^i`, where `x` ranges over the units of `K`, is
+   - `q-1` if `q-1 ∣ i`
+   - `0`   otherwise
+4. `finite_field.card`: The cardinality `q` is a power of the characteristic of `K`.
+   See `card'` for a variant.
 
-section polynomial
+## Notation
+
+Throughout most of this file, `K` denotes a finite field with `q` elements.
```

Comment:
I see what you mean about it being there already. I guess I interpreted `q` as notation to be used in comments and the like. The sentence at the start is good, but maybe a more forceful formulation for here would be:
"`q` is notation for the cardinality of `K`".

Reply:
Fixed